src/clib/pio_msg.c

```diff
@@ -1638,13 +1796,19 @@ int delete_att_handler(iosystem_desc_t *ios)
     if (!(name = malloc((namelen + 1) * sizeof(char))))
         return pio_err(ios, NULL, PIO_ENOMEM, __FILE__, __LINE__);
     if ((mpierr = MPI_Bcast(name, namelen + 1, MPI_CHAR, 0, ios->intercomm)))
+    {
+        free(name);
         return check_mpi2(ios, NULL, mpierr, __FILE__, __LINE__);
+    }
     LOG((2, "delete_att_handler namelen = %d name = %s ncid = %d varid = %d ",
          namelen, name, ncid, varid));
 
     /* Call the create file function. */
     if ((ret = PIOc_del_att(ncid, varid, name)))
+    {
+        free(name);
         return pio_err(ios, NULL, ret, __FILE__, __LINE__);
+    }
```

Comment:
I also do not know the reason why we use heap based memory for a name value. If we all agree to use a stack based one, we can get rid of free(name) everywhere, and do not have to change the existing code like:
```
if ((mpierr = MPI_Bcast(...))
    return check_mpi2(NULL, mpierr, __FILE__, __LINE__);
```